cloudflare/resource_cloudflare_access_mutual_tls_certificate.go

```diff
@@ -0,0 +1,195 @@
+}
+
+func resourceCloudflareAccessMutualTLSCertificateRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	identifier, err := initIdentifier(d)
+	if err != nil {
+		return err
+	}
+
+	var accessMutualTLSCert cloudflare.AccessMutualTLSCertificate
+	if identifier.Type == AccountType {
+		accessMutualTLSCert, err = client.AccessMutualTLSCertificate(identifier.Value, d.Id())
+	} else {
+		accessMutualTLSCert, err = client.ZoneAccessMutualTLSCertificate(identifier.Value, d.Id())
+	}
+
+	if err != nil {
+		if strings.Contains(err.Error(), "HTTP status 404") {
+			log.Printf("[INFO] Access Mutal TLS Certificate %s no longer exists", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error finding Access Mutual TLS Certificate %q: %s", d.Id(), err)
+	}
+
+	d.Set("name", accessMutualTLSCert.Name)
+	d.Set("associated_hostnames", accessMutualTLSCert.AssociatedHostnames)
+	d.Set("fingerprint", accessMutualTLSCert.Fingerprint)
+
+	return nil
+}
+
+func resourceCloudflareAccessMutualTLSCertificateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	updatedAccessMutualTLSCert := cloudflare.AccessMutualTLSCertificate{
+		ID:   d.Id(),
+		Name: d.Get("name").(string),
+	}
+	updatedAccessMutualTLSCert.AssociatedHostnames = expandInterfaceToStringList(d.Get("associated_hostnames"))
+
+	log.Printf("[DEBUG] Updating Cloudflare Access Mutal TLS Certificate from struct: %+v", updatedAccessMutualTLSCert)
+
+	identifier, err := initIdentifier(d)
+	if err != nil {
+		return err
+	}
+
+	var accessMutualTLSCert cloudflare.AccessMutualTLSCertificate
+	if identifier.Type == AccountType {
+		accessMutualTLSCert, err = client.UpdateAccessMutualTLSCertificate(identifier.Value, d.Id(), updatedAccessMutualTLSCert)
+	} else {
+		accessMutualTLSCert, err = client.UpdateZoneAccessMutualTLSCertificate(identifier.Value, d.Id(), updatedAccessMutualTLSCert)
+	}
+	if err != nil {
+		return fmt.Errorf("error updating Access Mutual TLS Certificate for %s %q: %s", identifier.Type, identifier.Value, err)
+	}
+
+	if accessMutualTLSCert.ID == "" {
+
+	}
+
+	return resourceCloudflareAccessMutualTLSCertificateRead(d, meta)
+}
+
+func resourceCloudflareAccessMutualTLSCertificateDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	certID := d.Id()
+
+	log.Printf("[DEBUG] Deleting Cloudflare Access Mutual TLS Certificate using ID: %s", certID)
+
+	identifier, err := initIdentifier(d)
+	if err != nil {
+		return err
+	}
+
+	if identifier.Type == AccountType {
+		err = client.DeleteAccessMutualTLSCertificate(identifier.Value, certID)
+	} else {
+		err = client.DeleteZoneAccessMutualTLSCertificate(identifier.Value, certID)
+	}
+	if err != nil {
+		return fmt.Errorf("error deleting Access Mutual TLS Certificate for %s %q: %s", identifier.Type, identifier.Value, err)
+	}
+
+	resourceCloudflareAccessMutualTLSCertificateRead(d, meta)
```

Comment:
for a `Delete` we don't usually call the `Read` but instead just `d.SetId("")` as we don't have a need to perform another HTTP GET to attempt to retrieve the state; instead, just remove it. i see you are handling the missing resource in the `Read` as well.